lib/connectors/mongodb/connector.rb

```diff
@@ -13,6 +13,8 @@
 module Connectors
   module MongoDB
     class Connector < Connectors::Base::Connector
+      PAGE_SIZE = 100
```

Comment:
We're currently not putting any connector-specific settings into YAML config, this can go to configurable fields or stay as a constant.

I'm not currently super sure about this, I could put it to configurable fields, but I'm a bit cautious about the growth of configurable fields.

Do you think it's worth exposing this value via UI?

Reply:
I'm honestly not sure about optimising PAGE_SIZE here, because it's a setting that depends both on host memory capabilities and on mongodb collection used. If collection is large, then this number can be reduced to take less memory. If the host is small, it has to be reduced more. If there are connector hosts of different size (which is not impossible), it's not possible to optimise the setting for all hosts any more, but only for the smallest ones.

Reply:
@seanstory my current worry is that our current configurable fields are not very user-friendly, and it would be hard to explain what this setting is without getting too deep into the explanations or just providing a bad user experience.

Reply:
personally, I'm not too worried about this, especially for a tech preview product. I think we can start by giving customers control, and then work on polishing the ux. But I'm curious about Serena's thoughts here too.

Reply:
It's a tech preview, so we can be a bit messy. We've already done that to a certain degree with our "direct connection (true/false)" configuration which no one really understands.  I totally agree with Sean that we can optimize for power here, and then refine the user experience once we see it in action.

Reply:
My 2 cents here: it should not goes to configurable fields. As an operator, I don't expect to configure this kind of low-level configs.

It can go to YAML under key `mongodb`. So we can make it a convention that developers can have connector-specific configs under the key `service_type`, but this will bring some challenges for Cloud deployment. One solution is that we can have a logic to copy all native connectors' (currently `mongodb` for ruby and `mysql` for python) configs from ent-search YAML. But that should be absolutely addressed in a separate task.

So the best we can do for this PR may be just hardcoding.

Reply:
After trying out the change that makes the batch_size configurable, I am also leaning to having this value as a constant as well. I'll revert the change that exposes it in the UI.

> It can go to YAML under key mongodb.
We removed configs for specific connectors from YAML in Ruby and I think it makes sense to keep it that way when moving forward - first because of the Cloud deployments, second because of our movement towards serverless.

We can, of course, allow it in future if we consider it to be the best way forward, but for now I'd limit it to a hardcoded value and follow-up later when needed.

Reply:
thanks for the follow up on what y'all decided